src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -206,6 +213,86 @@ def get_role_policy(self, role_name, policy_name: str | None = None, attached_po
                     policy_actions.append(AWSPolicyAction("s3", privilege, f"s3a://{match.group(1)}"))
         return policy_actions
 
+    def add_uc_role(self, role_name):
+        aws_role_trust_doc: dict = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"
+                    },
+                    "Action": "sts:AssumeRole",
+                    "Condition": {"StringEquals": {"sts:ExternalId": "0000"}},
+                }
+            ],
+        }
+        assume_role_json = self._get_json_for_cli(aws_role_trust_doc)
+        add_role = self._run_json_command(
+            f"iam create-role --role-name {role_name} --assume-role-policy-document {assume_role_json}"
+        )
+        if not add_role:
+            return False
+        return True
+
+    def add_uc_role_policy(self, role_name, policy_name, s3_prefixes: set[str], account_id: str, kms_key=None):
+        s3_prefixes_enriched = sorted([self.S3_PREFIX + s3_prefix for s3_prefix in s3_prefixes])
+        if kms_key:
+            policy_document = {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Action": [
+                            "s3:GetObject",
+                            "s3:PutObject",
+                            "s3:DeleteObject",
+                            "s3:ListBucket",
+                            "s3:GetBucketLocation",
+                        ],
+                        "Resource": s3_prefixes_enriched,
+                        "Effect": "Allow",
+                    },
+                    {
```

Comment:
The only difference between this if and else branch is the kms statement. Don't copy-paste entire policy, append a statement instead 

This will make it shorter

Reply:
Fixed